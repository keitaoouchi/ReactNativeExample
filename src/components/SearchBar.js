import React from "react";
import {Platform, TextInput, View} from "react-native";
import Icon from "react-native-vector-icons/FontAwesome";
import Color from "../misc/Color";
import Workaround from "../misc/Workaround";

export default SearchBar = (props) => (
  <View style={style.root}>
    <View style={style.search}>
      <Icon name="search" size={20} color="lightgray" style={style.icon}/>
      <TextInput
        style={style.input}
        onChangeText={props.onChange}
        onSubmitEditing={props.onSubmit}
        value={props.inputValue}
      />
    </View>
  </View>
);

const style = {
  root: {
    backgroundColor: Color.ivory,
    height: Workaround.style.root.marginTop,
    top: 0,
    left: 0,
    right: 0,
    position: 'absolute',
    flex: 1,
  },
  search: {
    flex: 1,
    ...Platform.select({
      ios: {
        top: 20,
      },
      android: {
        top: 8,
      },
    }),
    flexDirection: 'row',
    alignItems: 'flex-start',
  },
  icon: {
    marginTop: 8,
    marginBottom: 8,
    marginLeft: 8,
    flex: 1,
  },
  input: {
    flex: 12,
    ...Platform.select({
      ios: {
        height: 40,
      },
      android: {
        height: 44,
      },
    }),
    marginRight: 12,
    paddingLeft: 12,
    borderColor: Color.ivory,
    borderWidth: 1,
    borderRadius: 8,
    backgroundColor: Color.gray,
  },
};
