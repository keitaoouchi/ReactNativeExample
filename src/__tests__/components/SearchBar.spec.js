import "react-native";
import React from "react";
import renderer from "react-test-renderer";
import SearchBar from "../../../src/components/SearchBar";

describe('SearchBar', () => {

  it('renders correctly', () => {
    const tree = renderer.create(
      <SearchBar />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});