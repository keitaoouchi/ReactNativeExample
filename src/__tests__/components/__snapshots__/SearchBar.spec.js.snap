// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#EFEFF2",
      "flex": 1,
      "height": 64,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "flex": 1,
        "flexDirection": "row",
        "top": 20,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "lightgray",
            "fontSize": 20,
          },
          Object {
            "flex": 1,
            "marginBottom": 8,
            "marginLeft": 8,
            "marginTop": 8,
          },
          Object {
            "fontFamily": "FontAwesome",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      ï€‚
    </Text>
    <TextInput
      onChangeText={undefined}
      onSubmitEditing={undefined}
      style={
        Object {
          "backgroundColor": "#e6e6e7",
          "borderColor": "#EFEFF2",
          "borderRadius": 8,
          "borderWidth": 1,
          "flex": 12,
          "height": 40,
          "marginRight": 12,
          "paddingLeft": 12,
        }
      }
      value={undefined}
    />
  </View>
</View>
`;
